stdlib/public/SDK/Foundation/Data.swift

```diff
@@ -1082,6 +1082,43 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
         
     }
     
+    // slightly faster paths for common sequences
+    
+    public init<S: Sequence>(_ elements: S) where S.Iterator.Element == UInt8 {
+        let underestimatedCount = elements.underestimatedCount
+        self.init(count: underestimatedCount)
+        var idx = 0
+        for byte in elements {
```

Comment:
the other variants added here go through the withUnsafeBytes accessor to memcpy directly into the allocated backing buffer. So this is just the fallback case